crypto/kyber/p_kyber.c

```diff
@@ -29,6 +29,53 @@ static int pkey_kyber512_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
   return 1;
 }
 
+static int pkey_kyber512_encapsulate(EVP_PKEY_CTX *ctx,
+                                     uint8_t *ciphertext,
+                                     size_t  *ciphertext_len,
+                                     uint8_t *shared_secret,
+                                     size_t  *shared_secret_len) {
+  *ciphertext_len    = KYBER512_CIPHERTEXT_BYTES;
+  *shared_secret_len = KYBER512_SHARED_SECRET_BYTES;
+  if (ciphertext == NULL) { // Caller is performing a size check
```

Comment:
I guess the caller is really getting the parameter values, not performing a "check"?